### ISSUE TITLE
`as` is used too liberally within the crate

### ISSUE THREAD

**Issue by bzm3r** [StarArawn/bevy_ecs_tilemap#295]
We need to eliminate `as` (used for conversion between unsigned integers, integers, and floats) and restrict usage to truly lossless cases, or cases where truncation is desired behaviour. 

A major step towards this will be changing the shaders, so that they use  `u32` for tile positions.

**Comment by StarArawn** [comment]
> A major step towards this will be changing the shaders, so that they use `u32` for tile positions.

I agree with as probably being used to liberally but positions do need to be f32 in the shader, at least at the point that we transform them from local space to screen space.
